Kotlin/sort/InsertionSort/src/InsertionSort.kt

```diff
@@ -0,0 +1,32 @@
+fun insertionSort(array: MutableList<Int>) : List<Int> {
+    // go from the 2nd item in the list till the last item
+    for (i in 1 until array.size){
+        // store the value
+        val value = array[i]
+        // store the index of the value
+        var temp = i
+
+        // for every new_value below the index of the value
+        // if the new_value is greater than the value
+
+        // if the value is not greater than the loop will end
+        while ((array[temp - 1] > value) and (temp >= 0)) {
+            // put new_value one index above the current value
+            array[temp] = array[temp - 1]
+            // decrease the temporary index
+            temp -= 1
+        }
+        // when the loop has ended we know that the value at index
+        // less then current temp is lesser than current value so
+        // replace value at temp and close the loop
+        array[temp] = value
+    }
+    return array
+}
+
+fun main() {
+    val list = listOf(11,22,33,45,34,53,27,36)
```

Comment:
Make the input dynamic

Reply:
Do you mean to make it so that it accepts inputs from `stdin`

Reply:
@HarshCasper  i added the requested changes in a commit and pushed it. let me know if it is okay.